rose/server/score.py

```diff
@@ -27,54 +27,64 @@ def process(players, track):
+            # Move forlward
+            player.score += config.score_move_forward
 
-    for player in sorted_players:
-        player.score += config.score_move_forward
-        obstacle = track.get(player.x, player.y)
-        if obstacle == obstacles.CRACK:
-            if player.action != actions.JUMP:
-                track.clear(player.x, player.y)
-                player.y += 1
-                player.score += config.score_move_backward * 2
-            else:
-                player.score += config.score_jump
         elif obstacle in (obstacles.TRASH,
                           obstacles.BIKE,
                           obstacles.BARRIER):
-            if player.action not in (actions.LEFT, actions.RIGHT):
+            # Move back
+            track.clear(player.x, player.y)
+            player.y += 1
+            player.score += config.score_move_backward
+
+        elif obstacle == obstacles.CRACK:
+            if player.action == actions.JUMP:
+                # Move forlward
+                player.score += config.score_move_forward
+                player.score += config.score_jump
+            else:
+                # Move back
                 track.clear(player.x, player.y)
                 player.y += 1
-                player.score += config.score_move_backward * 2
+                player.score += config.score_move_backward
+
         elif obstacle == obstacles.WATER:
-            if player.action != actions.BRAKE:
+            if player.action == actions.BRAKE:
+                # Move forlward
+                player.score += config.score_move_forward
+                player.score += config.score_brake
+            else:
+                # Move back
                 track.clear(player.x, player.y)
                 player.y += 1
-                player.score += config.score_move_backward * 2
-            else:
-                player.score += config.score_brake
+                player.score += config.score_move_backward
+
         elif obstacle == obstacles.PENGUIN:
             if player.action == actions.PICKUP:
+                # Move forlward
                 track.clear(player.x, player.y)
                 player.score += config.score_move_forward
+                player.score += config.score_pickup
+            else:
+                # Move forlward no reward
+                player.score += config.score_move_forward
 
-        # Here we can end the game when player gets out of
-        # the track bounds. For now, just keep the player at the same
-        # location.
-        player.y = min(config.matrix_height - 1, max(2, player.y))
-
-        # Finally forget action
-        player.action = actions.NONE
+        # Filter for players with the same x and y
+        all_collisions = filter(lambda p:
+                                p.name != player.name and
+                                p.x == player.x and
+                                p.y == player.y, six.itervalues(players))
```

Comment:
I don't understand this code is doing - mostly likely nobody else will.

Reply:
removed in favor or reverting back to the sorting method